graph-builder/src/graph.rs

```diff
@@ -93,26 +112,9 @@ pub fn run(opts: &config::Options, state: &State) -> ! {
     }
 }
 
-fn create_graph(
-    opts: &config::Options,
-    username: Option<&str>,
-    password: Option<&str>,
-    cache: &mut HashMap<u64, Option<registry::Release>>,
-) -> Result<Graph, Error> {
+pub fn create_graph(releases: Vec<registry::Release>) -> Result<Graph, Error> {
```

Comment:
Carry-over comment from the other PR: if the input releases are both unique and order-insensitive, then the argument here is better typed as an `HashSet`.

Reply:
I had the same thoughts and then I went down the rabbit whole, which I just did again to give you the error output :-)

```
error[E0277]: the trait bound `std::collections::HashMap<std::string::String, std::string::String>: std::hash::Hash` is not satisfied
  --> graph-builder/src/release.rs:30:5
   |
30 |     pub metadata: HashMap<String, String>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::hash::Hash` is not implemented for `std::collections::HashMap<std::string::String, std::string::String>`
   |
   = note: required by `std::hash::Hash::hash`
```

I'm happy to use a `HashSet` if you have a proper solution for the above.

Reply:
Ah. I went through the same rabbit hole multiple times and still haven't found a satisfying answer. Leave it as it is then.